central/role/service/service_impl_test.go

```diff
@@ -454,3 +495,390 @@ func TestEffectiveAccessScopeForSimpleAccessScope(t *testing.T) {
+		s.Require().NoError(err)
+		clusterStore, err := clusterDataStore.GetTestPostgresDataStore(s.T(), s.postgres.Pool)
+		s.Require().NoError(err)
+		namespaceStore, err := namespaceDataStore.GetTestPostgresDataStore(s.T(), s.postgres.Pool)
+		s.Require().NoError(err)
+
+		s.service = &serviceImpl{
+			roleDataStore:      roleStore,
+			clusterDataStore:   clusterStore,
+			namespaceDataStore: namespaceStore,
+		}
+	} else {
+		s.boltEngine, err = boltPkg.NewTemp("roleServiceTestBolt")
+		s.Require().NoError(err)
+		s.rocksEngine, err = rocksdb.NewTemp("roleServiceTest")
+		s.Require().NoError(err)
+		s.bleveIndex, err = globalindex.MemOnlyIndex()
+		s.Require().NoError(err)
+
+		keyFence := dackboxConcurrency.NewKeyFence()
+		indexQ := queue.NewWaitableQueue()
+		dacky, err := dackbox.NewRocksDBDackBox(s.rocksEngine, indexQ, []byte("graph"), []byte("dirty"), []byte("valid"))
+		s.Require().NoError(err)
+
+		roleStore, err := roleDatastore.GetTestRocksBleveDataStore(s.T(), s.rocksEngine)
+		s.Require().NoError(err)
+		clusterStore, err := clusterDataStore.GetTestRocksBleveDataStore(s.T(), s.rocksEngine, s.bleveIndex, dacky, keyFence, s.boltEngine)
+		s.Require().NoError(err)
+		namespaceStore, err := namespaceDataStore.GetTestRocksBleveDataStore(s.T(), s.rocksEngine, s.bleveIndex, dacky, keyFence)
+		s.Require().NoError(err)
+
+		s.service = &serviceImpl{
+			roleDataStore:      roleStore,
+			clusterDataStore:   clusterStore,
+			namespaceDataStore: namespaceStore,
+		}
+	}
+}
+
+func (s *serviceImplTestSuite) TearDownSuite() {
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
+		s.postgres.Teardown(s.T())
+	} else {
+		s.Require().NoError(s.boltEngine.Close())
+		s.Require().NoError(rocksdb.CloseAndRemove(s.rocksEngine))
+		s.Require().NoError(s.bleveIndex.Close())
+	}
+}
+
+func (s *serviceImplTestSuite) SetupTest() {
+	s.storedClusterIDs = make([]string, 0)
+	s.storedNamespaceIDs = make([]string, 0)
+}
+
+func (s *serviceImplTestSuite) TearDownTest() {
+	writeCtx := sac.WithAllAccess(context.Background())
+	doneSignal := concurrency.NewSignal()
+	for _, clusterID := range s.storedClusterIDs {
+		s.NoError(s.service.clusterDataStore.RemoveCluster(writeCtx, clusterID, &doneSignal))
+	}
+	<-doneSignal.Done()
+	s.storedClusterIDs = s.storedClusterIDs[:0]
+	for _, namespaceID := range s.storedNamespaceIDs {
+		s.NoError(s.service.namespaceDataStore.RemoveNamespace(writeCtx, namespaceID))
+	}
+}
+
+const (
+	namespaceUUIDNamespace = "namespace"
+
+	clusterPermission             = "Cluster"
+	deploymentPermission          = "Deployment"
+	deploymentExtensionPermission = "DeploymentExtension"
+	integrationPermission         = "Integration"
+	namespacePermission           = "Namespace"
+	networkGraphPermission        = "NetworkGraph"
+	nodePermission                = "Node"
+	rolePermission                = "Role"
+)
+
+func getTestResourceScopeSingleNamespace(_ *testing.T, clusterID string, namespace string) *sac.TestResourceScope {
```

Comment:
super-nit: you can drop the `*testing.T` parameter here, since functions defined within `_test.go` files cannot be called within non-test files anyways:
```suggestion
func getTestResourceScopeSingleNamespace(clusterID string, namespace string) *sac.TestResourceScope {
```